src/main/java/mods/eln/misc/Recipe.java

```diff
@@ -23,8 +23,8 @@ public Recipe(ItemStack input, ItemStack output, double energy) {
     }
 
     public boolean canBeCraftedBy(ItemStack stack) {
-        if (stack.isEmpty()) return false;
-        return input.getCount() <= stack.getCount() && Utils.areSame(stack, input);
+        if (stack == null) return false;
```

Comment:
logically, it may be important to check if it's empty and if it's null. Not sure.